File: test-automation-plugin/examples/javascript-calculator/src/calculator.js
/**
 * Simple calculator module for demonstrating automated testing plugin.
 *
 * This module provides basic arithmetic operations and is designed to be used
 * as a test subject for the Automated Testing Plugin for Claude Code.
 */

/**
 * Add two numbers together.
 *
 * @param {number} a - First number
 * @param {number} b - Second number
 * @returns {number} The sum of a and b
 *
 * @example
 * add(2, 3)
 * // returns 5
 *
 * @example
 * add(-1, 1)
 * // returns 0
 *
 * @example
 * add(2.5, 3.7)
 * // returns 6.2
 */
function add(a, b) {
  return a + b;
}

/**
 * Subtract second number from first number.
 *
 * @param {number} a - Number to subtract from
 * @param {number} b - Number to subtract
 * @returns {number} The difference between a and b (a - b)
 *
 * @example
 * subtract(5, 3)
 * // returns 2
 *
 * @example
 * subtract(10, 15)
 * // returns -5
 *
 * @example
 * subtract(7.5, 2.5)
 * // returns 5.0
 */
function subtract(a, b) {
  return a - b;
}

/**
 * Multiply two numbers together.
 *
 * @param {number} a - First number
 * @param {number} b - Second number
 * @returns {number} The product of a and b
 *
 * @example
 * multiply(3, 4)
 * // returns 12
 *
 * @example
 * multiply(-2, 5)
 * // returns -10
 *
 * @example
 * multiply(2.5, 4)
 * // returns 10.0
 */
function multiply(a, b) {
  return a * b;
}

/**
 * Divide first number by second number.
 *
 * @param {number} a - Numerator
 * @param {number} b - Denominator (must not be zero)
 * @returns {number} The quotient of a divided by b
 * @throws {Error} If b is zero
 *
 * @example
 * divide(10, 2)
 * // returns 5.0
 *
 * @example
 * divide(7, 2)
 * // returns 3.5
 *
 * @example
 * divide(-10, 2)
 * // returns -5.0
 *
 * @example
 * divide(5, 0)
 * // throws Error: Cannot divide by zero
 */
function divide(a, b) {
  if (b === 0) {
    throw new Error("Cannot divide by zero");
  }
  return a / b;
}

module.exports = {
  add,
  subtract,
  multiply,
  divide,
};

File: test-automation-plugin/examples/python-calculator/src/calculator.py
"""
Simple calculator module for demonstrating automated testing plugin.

This module provides basic arithmetic operations and is designed to be used
as a test subject for the Automated Testing Plugin for Claude Code.
"""


def add(a, b):
    """
    Add two numbers together.

    Args:
        a: First number (int or float)
        b: Second number (int or float)

    Returns:
        The sum of a and b

    Examples:
        >>> add(2, 3)
        5
        >>> add(-1, 1)
        0
        >>> add(2.5, 3.7)
        6.2
    """
    return a + b


def subtract(a, b):
    """
    Subtract second number from first number.

    Args:
        a: Number to subtract from (int or float)
        b: Number to subtract (int or float)

    Returns:
        The difference between a and b (a - b)

    Examples:
        >>> subtract(5, 3)
        2
        >>> subtract(10, 15)
        -5
        >>> subtract(7.5, 2.5)
        5.0
    """
    return a - b


def multiply(a, b):
    """
    Multiply two numbers together.

    Args:
        a: First number (int or float)
        b: Second number (int or float)

    Returns:
        The product of a and b

    Examples:
        >>> multiply(3, 4)
        12
        >>> multiply(-2, 5)
        -10
        >>> multiply(2.5, 4)
        10.0
    """
    return a * b


def divide(a, b):
    """
    Divide first number by second number.

    Args:
        a: Numerator (int or float)
        b: Denominator (int or float, must not be zero)

    Returns:
        The quotient of a divided by b

    Raises:
        ValueError: If b is zero

    Examples:
        >>> divide(10, 2)
        5.0
        >>> divide(7, 2)
        3.5
        >>> divide(-10, 2)
        -5.0
        >>> divide(5, 0)
        Traceback (most recent call last):
        ...
        ValueError: Cannot divide by zero
    """
    if b == 0:
        raise ValueError("Cannot divide by zero")
    return a / b
